### DISCUSSION TITLE
POI AbstractPoiPersistenceManager.java - Deal properly with equal sign in tag values

### DISCUSSION THREAD

**Discussion by Juanjo-MC** [mapsforge/mapsforge#1279] [Ideas]
Hi,

in current implementation, `stringToTags` is not taking into account tags that contains an equal sign in the value. 
Such values are valid in osm and may appear ie on urls containing a query string. As an example, Nominatim is dealing properly with such values.

May I suggest to change the way the separation of tag/value is done to something like this?

```
String key = s.split(String.valueOf(Tag.KEY_VALUE_SEPARATOR))[0];
String value = s.substring(key.length() + 1);
```

Cheers

**Comment by devemux86** [comment]
We can probably do that.
Do you want to provide a pull request?

**Reply by Juanjo-MC** [reply]
Sure, here it is: https://github.com/mapsforge/mapsforge/pull/1280

I saw that `stringToTags` in PoiWriter.java is not using `KEY_VALUE_SEPARATOR`. Is that on purpose or is it an oversight?

**Reply by devemux86** [reply]
This is probably how it was written in the past.
